File: Arrays/BestTimeBuyAndSell/bestprofit.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

                 from sys import maxsize
                 lst = maxsize
                 op = 0
                 pist = 0
                 for i in range(len(prices)):
                     if prices[i] < lst :
                         lst = prices[i]
                         pist = prices[i] - lst 
                         if pist > op:
                             op = pist
                 return op

File: Arrays/BestTimeBuyAndSell/bestProfit.go
import "math"

func maxProfit(prices []int) int {
    lsf := math.MaxInt32  
    op := 0
    pist := 0
    
    for i := 0; i < len(prices); i++ {
        if prices[i] < lsf {
            lsf = prices[i]
        }
        pist = prices[i] - lsf
        if op < pist {
            op = pist
        }
    }
    return op  
}